cmd/vsphere-xcopy-volume-populator/vsphere-xcopy-volume-populator.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/rgolangh/xcopy-volume-populator/internal/ontap"
+	"github.com/rgolangh/xcopy-volume-populator/internal/populator"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+const (
+	prefix     = "forklift.konveyor.io"
+	mountPath  = "/mnt"
+	devicePath = "/dev/block"
+)
+
+var version = "unknown"
+
+var (
+	mode string
+	//secretRef is needed because of lack of control on the populator pod
+	// deployment. When/if we gain control on the pod deployment we should mount
+	// the secret as env vars. There is an attempt to push that,
+	// see https://github.com/kubernetes-csi/lib-volume-populator/pull/171
+	secretRef       string
+	sourceVMDKFile  string
+	targetPVC       string
+	targetNamespace string
+	storageVendor   string
+	storageHostname string
+	storageUsername string
+	storagePassword string
+	vsphereHostname string
+	vsphereUsername string
+	vspherePassword string
+
+	// kube args
+	httpEndpoint string
+	metricsPath  string
+	masterURL    string
+	kubeconfig   string
+
+	imageName   string
+	showVersion bool
+	namespace   string
+)
+
+func main() {
+
+	klog.InitFlags(nil)
+	// Main arg
+	flag.StringVar(&mode, "mode", "", "Mode to run in (controller, populate)")
+
+	// Populator args
+	flag.StringVar(&sourceVMDKFile, "source-vmdk", "", "File name to populate")
+	flag.StringVar(&targetPVC, "target-pvc", "", "Target PVC for population")
+	flag.StringVar(&targetNamespace, "target-namespace", "", "Contents to populate file with")
+	flag.StringVar(&storageVendor, "storage-vendor", "ontap", "The storage vendor to work with. Current values: [ontap,]")
+	flag.StringVar(&storageHostname, "storage-hostname", "", "The storage vendor api hostname")
+	flag.StringVar(&storageUsername, "storage-username", "", "The storage vendor api username")
+	flag.StringVar(&storagePassword, "storage-password", "", "The storage vendor api password")
+
+	flag.StringVar(&vsphereHostname, "vsphere-hostname", "", "vSphere's API hostname")
+	flag.StringVar(&vsphereUsername, "vsphere-username", "", "vSphere's API username")
+	flag.StringVar(&vspherePassword, "vsphere-password", "", "vSphere's API password")
```

Comment:
Please pass the passwords via file instead of args for security.

Reply:
i'm building on the populator pod to mount the secret as env params. this is not needed anymore and I'll remove it. I think secret-ref is also not needed anymore

Reply:
perhaps it is just needed for the populator controller to mount it. I'll look again